refactor: extract graphman core functions

YassinEldeeb opened the pull request.

lutter commented:
@YassinEldeeb Do you have the schema for the GraphQL server this generates somewhere? It would help in reviewing these changes.

lutter commented on core/src/graphman/core/create.rs:
```diff
@@ -0,0 +1,21 @@
+
+    if enable_logging {
+        println!("creating subgraph {}", name);
+    }
```
A better way to print conditionally would be to pass an argument `out: &dyn mut std::io::Write` to these methods and change all instances of `println!(..)` to `writeln!(out, ..)`

From the CLI, we'd pass in `std::io::stdout().lock()`, and from the API, we'd pass in a dummy implementation of `std::io::Write` that does nothing.

YassinEldeeb replied:
That's a pretty neat! I have implemented it, Thanks!

lutter reviewed (commented):
Not a full review, but a few things I noticed when scanning through the code.

Could you also try and rebase the PR on top of latest master? Can you also format the commit messages in accordance with [CONTRIBUTING.md](https://github.com/graphprotocol/graph-node/blob/master/CONTRIBUTING.md#commit-messages-and-pull-requests) ?

YassinEldeeb commented:
> @YassinEldeeb Do you have the schema for the GraphQL server this generates somewhere? It would help in reviewing these changes.

Hey! Yes we can emit them, but there's no graphql included in this PR, this only targets extracting graphman commands into core functions which can be later used by graphman API which is in this main PR #4598, the above is only a part of it.

YassinEldeeb commented:
> Could you also try and rebase the PR on top of latest master? Can you also format the commit messages in accordance with [CONTRIBUTING.md](https://github.com/graphprotocol/graph-node/blob/master/CONTRIBUTING.md#commit-messages-and-pull-requests) ?

I rebased and squashed my commits following the contribution guide.

YassinEldeeb commented on core/src/graphman/core/assign.rs:
```diff
@@ -0,0 +1,128 @@
+        "Waiting {}s to make sure pausing was processed",
+        sleep.as_secs()
+    );
+    thread::sleep(sleep);
```
This is not reached from the API, it had a sleep call in the graphman command, so I left it as is.
Should i remove it?

YassinEldeeb commented on core/src/graphman/core/assign.rs:
```diff
@@ -0,0 +1,128 @@
+            return Ok(());
+        }
+    };
+    println!("Operation completed");
```
`pause_or_resume` currently is only used in the CLI environment of graphman, there's no near plans of integrating it with graphman. After talking with the indexers, we've agreed that they only need those vital commands to be accessible via a graphql interface:
- List deployments
- Info deployment
- Drop deployment
- Set Target deployment
- Stats / account-like deployment
- Reassign deployment

YassinEldeeb commented on core/src/graphman/core/copy.rs:
```diff
@@ -0,0 +1,306 @@
+        DateTime<Utc>,
+        String,
+        Shard,
+    )>,
```
Yes. Agree! I'll add it.

YassinEldeeb commented on core/src/graphman/core/copy.rs:
```diff
@@ -0,0 +1,306 @@
+    //             None => status("queued", queued_at),
+    //         };
+    //     }
+    // }
```
It should be removed, since the logging logic has been integrated within the `list` command that extends the this core function in the graphman CLI.

Removed it!

azf20 commented:
hey @YassinEldeeb checking in on this?

lutter commented:
Closing this since it's not been active for a while. Feel free to reopen when you're ready to work on this again
